### ISSUE TITLE
[question] Referencing custom cmake macros in conan create?

### ISSUE THREAD

**Issue by nichmj01** [conan-io/conan#8482]
All of our projects have the same cmake setup so in an effort to simplify the top-level CMakeLists.txt file we have a series of macros in .cmake files that can be included to access helper macros. We distribute the source code and all cmake files to customers so that they can build the code themselves, and they need to be able to build the code without using conan. However,  we're trying to use conan internally to manage dependencies and to pull in pre-built binaries so that developers don't have to re-build projects individually. As such, the CMakeLists.txt files we provide must be usable without conan being present. Currently, the checks to switch between using conan or not is hidden inside the internal macros/.cmake files and controlled using a cmake variable.

I've exported the .cmake files as a cmake config package so that I can use find_package(buildtools) to find the needed macros in each project. This can be done at the top of the CMakeLists.txt file and from then our macros can be used to manage where conan is used or not. This works great when I do the build steps manually from the command line: conan install, cmake configure, cmake build since I can specify the location of the buildtools-config.cmake file in the CMAKE_PREFIX_PATH.

Now I want to be able to use conan create to automatically build the source and create conan packages but find_package(buildtools) fails during the conan build/cmake configure step. I created a buildtools conan package which contains all the .cmake files and added it a build requirement to projects that need it, but haven't been able to get it to work. 

Here is the layout for the buildtools conan package:

package_root
+ lib
  + buildtools
    + cmake
       + internal
          + buildtools.cmake
       + buildtools-config.cmake (this essentially has the line include(internal/buildtools.cmake)
       + buildtools-version.cmake

I figured I would follow the cmake package install structure, but not sure if this is right.

Here is what I've tried so far:

Using the 'cmake' generator doesn't seem like a good fit since I would need to include the conanbuildinfo.cmake file in the CMakeLists.txt which I can't do since conan is not always used. Once find_package(buildtools) completes I can use this but it can't be used to help figure out where to look for the buildtools package.

Using the 'cmake_paths' generator seems like it could work since it would add the path to the buildtools conan package in the cache to the CMAKE_PREFIX_PATH to allow buildtools-config.cmake to be found but I can't figure out how to make conan use the generated conan_paths.cmake in the build step. If I manually include the conan_paths.cmake then it'll work but that file is not always guaranteed to be there.

Using the 'cmake_find_package_multi' generator creates it's own buildtools-config.cmake file which lets find_package work but the generated -config.cmake file doesn't include the macros so cmake configure fails later on. 

I've also started messing round with the updated cmake tools in conan.tools.cmake and with CMakeToolchain, which automagically adds the -DCMAKE_TOOLCHAIN_FILE="conan_toolchain.cmake" line to cmake but that file don't seem to reference the locations in the conan cache.

Any thoughts? 

I'm okay with adding the generated -config.cmake files with my conan packages so maybe all I really need is help with is getting the cmake_paths.cmake file added as a CMAKE_TOOLCHAIN_FILE, but I'm open to other/better options. I guess I could run the cmake commands manually? ie, not using the CMake Helper?

I'm using conan 1.33.

- [X ] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by jgsogo** [comment]
Hi, @nichmj01. One hint that might help:

 * Use [`build_modules` in the `cpp_info`](https://docs.conan.io/en/latest/reference/conanfile/attributes.html?highlight=build_modules#cpp-info) of your `buildtools` package to tell Conan that there are a list of files you want to include whenever someone consumes your package.
   
   ```python
   class Recipe(ConanFile):
      name = 'buildtools'

      def package_info(self):
         ...
         self.cpp_info.build_modules['cmake_find_package'] = [... list of '.cmake' files you want to include ...]
         self.cpp_info.build_modules['cmake_find_package_multi'] = [... list of '.cmake' files you want to include ...]
   ```

   Now, if you inspect the generated `Findbuildtools.cmake` or `buildtools-config.cmake` file generated by Conan you will see that Conan includes automatically the files listed. I'm intentionally using these generators (specially the `_multi` one) as they are the ones that will make the transition to Conan v2.0 easier.

Please, let me know if this is enough to achieve your objective, if not, I'll try to reproduce the scenario with actual code and I'm sure we will manage to provide a solution.

Thanks!

**Comment by nichmj01** [comment]
Thank jgsogo! This worked as you explained and addressed my question. 

I ran into different issues due to the way one of our projects is structured; we end up with multiple cmake packages but one conan package which means the cmake_find_package generators don't create a 1-to-1 equivalent to the cmake generated config.cmake files when not using conan. I suspect I could make it all work but at this point it's not worth the extra effort. I was able to get the cmake_paths approach to work well enough for now, although it sounds like that won't be supported in v2. We'll deal with that when we get there.
